beastling/sections.py

```diff
@@ -93,3 +154,133 @@ def __attrs_post_init__(self):
+        "multiple datasets with non-equal language sets.",
+    )
+    starting_tree = opt(
+        None,
+        "A starting tree in Newick format, or the name of a file containing the same.",
+        getter=functools.partial(get_tree, 'starting'))
+    sample_branch_lengths = opt(
+        True,
+        "A boolean value, controlling whether or not to estimate tree branch lengths.",
+        getter=ConfigParser.getboolean)
+    sample_topology = opt(
+        True,
+        "A boolean value, controlling whether or not to estimate tree topology.",
+        getter=ConfigParser.getboolean)
+    subsample_size = opt(
+        0,
+        'Number of languages to subsample from the set defined by the dataset(s) and other '
+        'filtering options like "families" or "macroareas".',
+        getter=ConfigParser.getint)
+    monophyly = opt(
+        False,
+        "A boolean parameter, controlling whether or not to enforce monophyly constraints derived "
+        "from Glottolog's classification.",
+        getter=ConfigParser.getboolean)
+    monophyletic = opt(
+        None,
+        "Backwards compat",
+        getter=ConfigParser.getboolean)
+    monophyly_start_depth = opt(
+        0,
+        "Integer; Starting depth in the Glottlog classification hierarchy for monophyly "
+        "constraints.",
+        getter=ConfigParser.getint)
+    monophyly_end_depth = opt(
+        None,
+        "Integer; Number of levels of the Glottolog classification to include in monophyly "
+        "constraints.",
+        getter=ConfigParser.getint)
+    monophyly_levels = opt(
+        sys.maxsize,
+        "Integer; Number of levels of the Glottolog classification to include in monophyly "
+        "constraints.",
+        getter=ConfigParser.getint)
+    monophyly_newick = opt(
+        None,
+        "Either a Newick tree string or the name of a file containing a Newick tree string which "
+        "represents the desired monophyly constraints if a classification other than Glottolog is "
+        "required.",
+        getter=functools.partial(get_tree, 'monophyly'))
+    monophyly_direction = opt(
+        "top_down",
+        "Either the string 'top_down' or 'bottom_up', controlling whether 'monophyly_levels' "
+        "counts from roots (families) or leaves (languages) of the Glottolog classification.",
+        converter=lambda s: s.lower())
+    tree_prior = opt(
+        "yule",
+        "Tree prior. Can be overridden by calibrations.")
+
+    def __attrs_post_init__(self):
+        # Now that languages are loaded, we can sanitise the trees:
+        if self.starting_tree:
+            self.starting_tree = sanitise_tree(self.starting_tree, 'starting', self.languages)
+        if self.monophyly_newick:
+            self.monophyly_newick = sanitise_tree(
+                self.monophyly_newick, 'monophyly', self.languages)
+        if self.monophyletic is not None:
+            self.monophyly = self.monophyletic
```

Comment:
Trees sanitized, backwards compat established - after initialization we can return a perfect config section!